### PR TITLE
Add instances for conv_scale with fp8 in/out

### PR DESCRIPTION
**Pull request by geyyer**
- Add fp8@fp8->fp8 instances
- Add example
- Add profiler mode
- Add client example

### CODE REVIEW

File: client_example/16_convnd_fwd/conv3d_fwd_fp8.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline client_example/16_convnd_fwd/conv3d_fwd_fp8.cpp]:
```suggestion
// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.
```

File: example/09_convnd_fwd/convnd_fwd_common.hpp [hunk 2]
```diff
@@ -27,6 +27,88 @@ void print_helper_msg()
               << ck::utils::conv::get_conv_param_parser_helper_msg() << std::endl;
 }
 
+template <typename DataType>
+inline __host__ __device__ constexpr double get_rtol()
+{
+    if constexpr(std::is_same_v<DataType, float>)
```
Comment (bartekxk) [inline example/09_convnd_fwd/convnd_fwd_common.hpp:33]: Can We change it to more generic? I propose to calculate it in this way:
`C * Z * Y * X * 2^-mantisa`
For example for 3 multiple adds:
```
(((x1*x2)*(1+eps) + (x3*x4))(1+eps) + (x5+6)(1+eps))(1+eps)= 
((x1 *x2 + x1*x2 * eps + x3 * x4 + x3*x4*eps) (1+eps)) + x5*x6 + x5*x6*eps)(1+eps) = 
(x1 *x2 + x1*x2 * eps + x3 * x4 + x3*x4*eps + x1 *x2  * eps+ x1*x2 * eps * eps + x3 * x4 * eps + x3*x4*eps * eps  + x5*x6+ x5*x6*eps)(1+eps)
```
We can skip eps*eps, it is very very low, then:
```
(x1 *x2 + x1*x2 * eps + x3 * x4 + x3*x4*eps + x1 *x2  * eps+  x3 * x4 * eps +  x5*x6+ x5*x6*eps)(1+eps) = 
x1 *x2 + x1*x2 * eps + x3 * x4 + x3*x4*eps + x1 *x2  * eps+  x3 * x4 * eps +  x5*x6+ x5*x6*eps + x1 *x2 * eps + x1*x2 * eps * eps + x3 * x4 + x3*x4*eps * eps + x1 *x2  * eps * eps+  x3 * x4 * eps * eps +  x5*x6+ x5*x6*eps * eps
```
Skip eps* eps again:
```
x1 *x2 + x1*x2 * eps + x3 * x4 + x3*x4*eps + x1 *x2  * eps+  x3 * x4 * eps +  x5*x6+ x5*x6*eps + x1 *x2 * eps = 
x1 +x2  +x3*x4 +x5*x6 + 3*x1*x2*eps + 2 x3*x4 eps + x5 *x6*eps
```
So for 3 multiple + add maximum  it should not exceed `3*x1*x2*eps + 2 x3*x4 eps + x5 *x6*eps` which should not exceed `3 * 2^-mantisa`.
For each convolution output cell we have `C * Z * Y * X` multiple+add.

For integers it should be accurate.

File: example/09_convnd_fwd/convnd_fwd_xdl_fp8.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline example/09_convnd_fwd/convnd_fwd_xdl_fp8.cpp]: Could You fix each Copyright header? For new files just 2024, for updated -2024.

**Review by bartekxk — COMMENTED** [review]
LGTM, just minor comments

File: client_example/16_convnd_fwd/conv3d_fwd_fp8.cpp (hunk 1 above)
Comment (geyyer) [reply client_example/16_convnd_fwd/conv3d_fwd_fp8.cpp]: Fixed in https://github.com/ROCm/composable_kernel/pull/1193/commits/27271089812dba5cca0c922bd48da42a67326a03

File: example/09_convnd_fwd/convnd_fwd_common.hpp (hunk 2 above)
Comment (geyyer) [reply example/09_convnd_fwd/convnd_fwd_common.hpp:33]: Hi @bartekxk, thanks for the detailed explanation! I like the idea to make acceptable tolerances meaningful, but I would like to clarify couple more details:
1. The logic you are suggesting is for relative tolerance calculation. We could also calculate acceptable absolute tolerance for various datatypes, do you think it is worth it?
2. Problem dimensions are runtime arguments, so we cannot calculate tolerances in compile-time then. I don't think the runtime overhead can become significant, and it is not kernel-related, but there will be some change.
3. From the low precision type perspective, there are several 8-bit float standards we are going to support, so calculating exact tolerances would make maintenance harder. For example, for various architectures we will have to add macros and calculate corresponding tolerances.

Looking forward to hear your thoughts! Overall, I'm totally for updating our tolerance logic repo-wide and I believe the only way to do it is one operator at a time ;-)

File: example/09_convnd_fwd/convnd_fwd_xdl_fp8.cpp (hunk 3 above)
Comment (geyyer) [reply example/09_convnd_fwd/convnd_fwd_xdl_fp8.cpp]: Fixed in https://github.com/ROCm/composable_kernel/commit/27271089812dba5cca0c922bd48da42a67326a03

File: example/09_convnd_fwd/convnd_fwd_common.hpp (hunk 2 above)
Comment (bartekxk) [reply example/09_convnd_fwd/convnd_fwd_common.hpp:33]: 1. Yes, this is for relative, the same example for absolute:
`((((x1*x2 + eps)  + (x3*x4 + eps)) + eps) + (x5*x6 +eps)) + eps = 
((x1*x2 + x3 * x4 + 3* eps) + (x5*x6 + eps) )+eps  =
x1 *x2 + x3 * x4 + x5*x6 + 5 * eps`
So we can assume that it is `2 * multiple add operations * 0.5 * ULP =  2 * C * Z * Y * X * 0.5 * ULP`, It is a little bit harder to specify ULP for each temporary value during calculation. One idea is to use maximum value from GenerateTensorValue. We can discuss it. 
2. Yes, but in this case we can just assume relative error equal to `C * Z * Y * X * 2^-mantisa`. Calculation of this formula should be very fast.
3. Maybe we can create some custom function like `ck::epsilon<DataType>()` to return 2^-mantisa for each DataType.

This change could be also done in seperate change which fix this calculation in whole repository :). I will be happy to help you implement this.

Comment (bartekxk) [reply example/09_convnd_fwd/convnd_fwd_common.hpp:33]: We agreed to implement such threshold calculation in next PR.

**Review by bartekxk — APPROVED** [review]

**Review by illsilin — APPROVED** [review]

**Pull request merged**